ldk-server/src/util/logger.rs

```diff
@@ -0,0 +1,123 @@
+use std::io::{self, BufWriter, Write};
+use std::path::Path;
+use std::sync::Mutex;
+
+/// A logger implementation that writes logs to both stderr and a file.
+///
+/// The logger formats log messages with RFC3339 timestamps and writes them to:
+/// - stdout/stderr for console output
+/// - A file specified during initialization
+///
+/// All log messages follow the format:
+/// `[TIMESTAMP LEVEL TARGET] MESSAGE`
+///
+/// Example: `[2025-12-04T10:30:45Z INFO ldk_server] Starting up...`
+pub struct ServerLogger {
+	/// The minimum log level to display
+	level: LevelFilter,
+	/// The file to write logs to, buffered and protected by a mutex for thread-safe access
+	file: Mutex<BufWriter<File>>,
+}
+
+impl ServerLogger {
+	/// Initializes the global logger with the specified level and file path.
+	///
+	/// Opens or creates the log file at the given path. If the file exists, logs are appended.
+	/// If the file doesn't exist, it will be created along with any necessary parent directories.
+	///
+	/// This should be called once at application startup. Subsequent calls will fail.
+	pub fn init(level: LevelFilter, log_file_path: &Path) -> Result<(), io::Error> {
+		// Open or create the log file
+		let file = if log_file_path.exists() {
+			OpenOptions::new().append(true).open(log_file_path)?
+		} else {
+			// Create parent directories if they don't exist
+			if let Some(parent) = log_file_path.parent() {
+				fs::create_dir_all(parent)?;
+			}
+			File::create(log_file_path)?
+		};
+
+		let buffered_file = BufWriter::new(file);
+		let logger = ServerLogger { level, file: Mutex::new(buffered_file) };
+
+		log::set_boxed_logger(Box::new(logger))
+			.map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+		log::set_max_level(level);
+		Ok(())
+	}
+}
+
+impl Log for ServerLogger {
+	fn enabled(&self, metadata: &Metadata) -> bool {
+		metadata.level() <= self.level
+	}
+
+	fn log(&self, record: &Record) {
+		if self.enabled(record.metadata()) {
+			let timestamp = format_timestamp();
+			let level_str = format_level(record.level());
+
+			// Log to console
+			let _ = match record.level() {
+				Level::Error => {
+					writeln!(
+						io::stderr(),
+						"[{} {} {}] {}",
+						timestamp,
+						level_str,
+						record.target(),
+						record.args()
+					)
+				},
+				_ => {
+					writeln!(
+						io::stdout(),
+						"[{} {} {}] {}",
+						timestamp,
+						level_str,
+						record.target(),
+						record.args()
+					)
+				},
+			};
+
+			// Log to file
+			if let Ok(mut file) = self.file.lock() {
+				let _ = writeln!(
+					file,
+					"[{} {} {}] {}",
+					timestamp,
+					level_str,
+					record.target(),
+					record.args()
+				);
+			}
+		}
+	}
+
+	fn flush(&self) {
+		let _ = io::stdout().flush();
+		let _ = io::stderr().flush();
+		if let Ok(mut file) = self.file.lock() {
+			let _ = file.flush();
+		}
+	}
+}
+
+fn format_timestamp() -> String {
+	let now = chrono::Utc::now();
+	now.to_rfc3339_opts(chrono::SecondsFormat::Secs, true)
```

Comment:
For debugging purposes, millis/micros can be useful, in particular when consuming logs from multiple nodes in a single stream. Maybe it's worth the additional clutter?